socks/socks4.py

```diff
@@ -39,10 +49,17 @@ def dumps(self) -> bytes:
 class SOCKS4Reply(typing.NamedTuple):
     reply_code: SOCKS4ReplyCode
     port: int
-    addr: bytes
+    addr: str
 
-    def loads(self, data: bytes) -> "SOCKS4Reply":
-        raise NotImplementedError()
+    @classmethod
+    def loads(cls, data: bytes) -> "SOCKS4Reply":
+        if len(data) != 8:
+            raise ProtocolError("Malformed reply")
+        return cls(
+            reply_code=SOCKS4ReplyCode(data[1:2]),
+            port=int.from_bytes(data[2:4], byteorder="big"),
+            addr=decode_address(AddressType.IPV4, bytes(data[4:8])),
```

Comment:
Don't need the `bytes()` call here

Reply:
I was getting `TypeError: unhashable type: 'bytearray'` exception because I was passing the recieved data bytesarray directly from the connection, interestingly mypy did not error on the type mismatch.